### DISCUSSION TITLE
Build Errors

### DISCUSSION THREAD

**Discussion by jstrahl3** [NoahGWood/OpenEFT#15] [General]
Good evening,
I'm following your instructions for the Linux install, but keep getting some errors when trying to do the build step.  I'm running Linux Mint and am by no means a Linux expert so it's totally possible I'm missing something.  I've attached screenshots of what I'm getting. The error in screenshot 1 occurs first and I'm prompted for credentials.  It runs a few more lines and then kicks me out with the final error in screenshot 2.  Thanks and I really appreciate what you're building here!

![build_screenshot_1](https://user-images.githubusercontent.com/128939943/227746552-00a0836d-3aa4-4ec0-939e-5fc8c1da7534.png)

![build_error_2](https://user-images.githubusercontent.com/128939943/227746564-60a6ee40-364e-41ab-9234-545ceaa15f55.png)

**Comment by NoahGWood** [comment]
Looks like you're missing the requests_cache package, you can install that using

pip install requests-cache

I'll make a note that we need to update the requirements.txt, there might be others missing too, so I'll double check here

**Comment by jstrahl3** [comment]
Thanks for the quick reply.  That did get me a little further, but I was still getting the backports.zoneinfo error.  A quick google search told me I needed to change the requirements.txt to backports.zoneinfo==0.2.1;python_version<"3.9".   That now gets me to hanging at this.

![build_screenshot_4](https://user-images.githubusercontent.com/128939943/227749213-5e132117-abf4-4619-921c-74e61f1473a5.png)

**Reply by tjordan170** [reply]
Just open your browser and go to 127.0.0.1:7100.

**Reply by jstrahl3** [reply]
Ahh yeah, I'm an idiot.  Thought it was going to open on it's own.  Looks to be working now.  Thanks for all the help!

**Reply by jstrahl3** [reply]
Spoke a little too soon.  I can make it all the way to the end with everything looking good, then after clicking "Submit" it generates this error.

![submit_error](https://user-images.githubusercontent.com/128939943/227750317-596d1de2-975a-4762-807a-886ec6d351a4.png)

**Comment by tjordan170** [comment]
Did you build the nbis binaries? You'll need to do that, then add the locations of nfseg and nfiq to your path.

Download the software [here](https://www.nist.gov/itl/iad/image-group/products-and-services/image-group-open-source-server-nigos), and reference the installation instructions [here](https://nvlpubs.nist.gov/nistpubs/Legacy/IR/nistir7392.pdf).

**Reply by jstrahl3** [reply]
I did not.  I've been working on that since yesterday and I've hit another wall that I can't seem to find a way around.  This is after running "make it" and the error is "multiple definition of 'histo-head'".  Any ideas here?  I really appreciate all the help here.

![make_it_error](https://user-images.githubusercontent.com/128939943/227943956-9d740f5b-57e8-4db1-8759-69cfbd28464e.png)

**Reply by NoahGWood** [reply]
Hmm... I'd try running make clean and starting over again

**Comment by jstrahl3** [comment]
I just figured it out.  I had to install an older version of gcc to get the thing to compile.  After that everything worked like a charm!  Now to submit to the AFT.

For future documentation here's the instructions I followed and the version of gcc I installed was 9 as it was the oldest that could be found using the method in the link.

https://stackoverflow.com/questions/63593100/how-to-install-older-version-of-gcc-3-or-gcc-4-in-ubuntu-20
